catroid/src/main/java/org/catrobat/catroid/utils/ImportProjectsFromExternalStorage.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.pm.ApplicationInfo;
+import android.os.AsyncTask;
+import android.preference.PreferenceManager;
+import android.support.v7.app.AlertDialog;
+import android.util.Log;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.FlavoredConstants;
+import org.catrobat.catroid.io.StorageOperations;
+import org.catrobat.catroid.ui.recyclerview.util.UniqueNameProvider;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.catrobat.catroid.common.FlavoredConstants.OLD_EXTERNAL_STORAGE_ROOT_DIRECTORY;
+import static org.catrobat.catroid.common.SharedPreferenceKeys.SHOW_COPY_PROJECTS_FROM_EXTERNAL_STORAGE_DIALOG;
+import static org.catrobat.catroid.io.StorageOperations.transferData;
+
+public class ImportProjectsFromExternalStorage implements DialogInterface.OnClickListener {
+	public static final String TAG = ImportProjectsFromExternalStorage.class.getSimpleName();
+	private static String pocketCodeEsFolderName = "Pocket Code";
+	private static String pocketCodeIsFolderName = "files";
+	private static String backpackFolderName = "backpack";
+	private Activity activity;
+
+	public ImportProjectsFromExternalStorage(Activity mActivity) {
+		activity = mActivity;
+	}
+
+	private static File copyProjectsFromExternalToInternalStorage(File srcDir, File dstDir) throws IOException {
```

Comment:
this is actually copying folders, not only Projects, and it doenst care if its internal or external storage since its getting just Files as src and destination.
copyFolderRecursive would be more fitting name for this.

Reply:
done